examples/microwave-oven-app/microwave-oven-common/include/microwave-oven-device.h

```diff
@@ -31,10 +31,10 @@ class MicrowaveOvenDevice
     {
         // set callback functions
         mMicrowaveOvenControlDelegate.SetMicrowaveOvenControlSetCookingParametersCallback(
-            std::bind(&MicrowaveOvenDevice::HandleMicrowaveOvenSetCookingParametersCallback, this, std::placeholders::_1,
+            std::bind(&MicrowaveOvenDevice::HandleMicrowaveOvenSetCookingParametersCommandCallback, this, std::placeholders::_1,
```

Comment:
This is much too complex. All of this could be done with a single delegate API that an implementer can implement and set. Having so many functor callbacks does not help make the microwave easier to implement. I expected a few interfaces to be implemented by whomever is doing the microwave, and whatever implementation of that class can talk to the actual hardware. Splitting it into many callbacks set into a delegate just adds one more level of indirection that makes things more complicated to setup.

Overall: the idea is that the common code from the SDK has to be able to interface with code that comes from applications which are OUTSIDE the SDK and so the cluster logic cannot know in advance the exact implementatoon. With the method of attribute change callbacks, this is done with global functions (which don't scale well). With the approach of most appliance code so far, there are delegates registered against the SDK common code (in the cluster server's logic). The common code in SDK calls into delegates which can be provided by external applications.

Moving to many callbacks just makes it harder to implement the product since in the end, there is likely a single module of logic that will receive all callbacks and mutate the common state of the microwave.